bgpd/bgp_rpki.c

```diff
@@ -905,20 +905,39 @@ static void free_cache(struct cache *cache)
 	XFREE(MTYPE_BGP_RPKI_CACHE, cache);
 }
 
+
+/* return true if config changed from default */
+static bool config_changed(void)
+{
+
+	if (polling_period != POLLING_PERIOD_DEFAULT)
+		return true;
+	if (retry_interval != RETRY_INTERVAL_DEFAULT)
+		return true;
+	if (expire_interval != EXPIRE_INTERVAL_DEFAULT)
+		return true;
+	if (listcount(cache_list))
+		return true;
+	return false;
+}
+
 static int config_write(struct vty *vty)
 {
 	struct listnode *cache_node;
 	struct cache *cache;
 
-	if (listcount(cache_list)) {
+	if (config_changed()) {
```

Comment:
config_changed() is done to introduce move to vrf contexts.
but whatever.
I modified slightly code so that listcount() is well readable, but config_changed is introduced to factorise and to be used later, when vrf context is ready.